superset/db_engine_specs/presto.py

```diff
@@ -1224,14 +1225,20 @@ def expand_data(  # pylint: disable=too-many-locals
         return all_columns, data, expanded_columns
 
     @classmethod
-    def extra_table_metadata(
-        cls, database: Database, table_name: str, schema_name: str | None
+    def get_extra_table_metadata(
+        cls,
+        database: Database,
+        table: Table,
     ) -> dict[str, Any]:
         metadata = {}
 
-        if indexes := database.get_indexes(table_name, schema_name):
+        if indexes := database.get_indexes(table.table, table.schema):
             col_names, latest_parts = cls.latest_partition(
-                table_name, schema_name, database, show_first=True, indexes=indexes
+                table.table,
```

Comment:
Should we be passing a `Table` instance everywhere?

Reply:
We should! I'm doing that in the next PR! 😊